src/toolchain.rs

```diff
@@ -119,4 +175,35 @@ impl Toolchain {
 
         Ok(download_cfg)
     }
+
+    pub fn remove_component(&self, component: &str) -> Result<()> {
+        if self.has_component(component) {
+            info!("Removing '{}' from toolchain '{}'", component, self.name);
+            let component_path = self.path.join(component);
+            remove_file(component_path)?;
+            // If component to remove is 'forc', silently remove forc plugins
+            if component == component::FORC {
+                for component in [
+                    component::FORC_FMT,
+                    component::FORC_LSP,
+                    component::FORC_EXPLORE,
+                ] {
+                    let component_path = self.path.join(component);
+                    remove_file(component_path)?;
```

Comment:
Same here

Reply:
Also - are we sure these plugins will always exist? E.g. in the near future we may add `FORC_WALLET` or `FORC_CLIENT` to this list. Perhaps we should first check `component_path.exists()`?

Reply:
in the entrypoint of `remove_component` we first check that `self.has_component`, which should handle checking that the path exists. So this shouldn't be a worry